adventofcode_3/index.js

```diff
@@ -0,0 +1,50 @@
+const fs = require("fs");
+
+const arrayOfInput = fs
+  .readFileSync(`${__dirname}/input.txt`)
+  .toString()
+  .split("\n")
+  .map(x => x.split(""))
+;
+
+function renderArray(array) {
```

Comment:
This code only works with 2d array...

For example, this code will not work in that case:

```
const array1d = ["a"];
// you will have an error like "join is not function"
renderArray(array1d);
```

**Some way of improvements:**

- You can help usage by renaming the arg `array` to `array2d`.
- You can add [jsdoc](https://jsdoc.app/about-getting-started.html):
- You can `assert` in order to prevent mi-usage

In rendering function, you should focus on forging a string and let the `console.log` outside for a better re-usability. Usually, output can change... right now you write your output in the `stdout` but you can also write in a file, in an http response, etc. 

## Sample using jsdoc

```js
/**
 * Render a 2 dimensional array
 * @param {Array<Array>} array 2 dimensional array to render
 */
function renderArray(array) { /** etc. */ }
```

## Sample using `assert`

```js
function renderArray(array) {
assert(Array.isArray(array), "array param must of type `array`");

return array.map(item => {
  assert(Array.isArray(item), "item of array must be of type array");
  return item.join("");
}).join("\n");
}

// sample usage
console.log(renderArray([[]]));
```

## Sample splitted version

```js
function render2dArray(toRender = [[]]) {
  return toRender.map(x => x.join("")).join("\n");
}

/**
 * Write to the stdout
 * @param {string} str String to write
 */
function stdoutWriter(str) {
  console.log(str);
}

/**
 * Sync write to a file
 * @param {string} str String to write
 * @param {{ filename: string }} options Extra options
 */
function fileWriter(str, options = { filename: "stdout.txt" }) {
  return fs.writeFileSync(options, str);
}

// sample usage
const array2d = [[".", "#"],["#","."]];

stdoutWriter(render2dArray(array2d));
// or just use `console.log` ... because overengineering is bad ;)
```